inject/src/main/java/io/micronaut/context/DefaultBeanContext.java

```diff
@@ -1574,9 +1574,6 @@ protected void registerConfiguration(@Nonnull BeanConfiguration configuration) {
                 if (beanFactory instanceof ParametrizedBeanFactory) {
                     ParametrizedBeanFactory<T> parametrizedBeanFactory = (ParametrizedBeanFactory<T>) beanFactory;
                     Argument<?>[] requiredArguments = parametrizedBeanFactory.getRequiredArguments();
-                    if (argumentValues == null) {
```

Comment:
Couldn't this just be changed to `if (argumentValues == null && requiredArguments.length > 0)` instead of reassigning the argument values and creating an optional?

Reply:
If the parameter is marked as `@Nullable` it will still be in requiredArguments list so additional check `requiredArguments.length > 0`  does not solve anything.